### PR TITLE
Doc: specify maximum allowed number of outstanding futures per PE

### PR DESCRIPTION
**Pull request by ZwFink**
Documented the limits on the number of outstanding futures per PE. Feel free to make suggestions on language, clarity, etc.

### CODE REVIEW

File: doc/charm++/manual.rst [hunk 1]
```diff
@@ -5858,6 +5858,15 @@ Other functions complete the API for futures. *CkReleaseFuture* destroys
 a future. *CkProbeFuture* tests whether the future has already finished
 computing the value of the expression.
 
+The maximum number of outstanding futures a PE may have is limited by the size of
+*CMK_REFNUM_TYPE*. Specifically, no more than :math:`2^{SIZE}-1`, where :math:`SIZE`
+is the size of *CMK_REFNUM_TYPE* in bits, may be outstanding at any time.
+Waiting on more futures will cause a fatal error in non-production builds,
+and will cause the program to hang in production builds. To overcome this limit,
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst:5865]:
```suggestion
and will cause the program to hang in production builds. To increase this limit,
```

File: doc/charm++/manual.rst [hunk 2]
```diff
@@ -5858,6 +5858,15 @@ Other functions complete the API for futures. *CkReleaseFuture* destroys
 a future. *CkProbeFuture* tests whether the future has already finished
 computing the value of the expression.
 
+The maximum number of outstanding futures a PE may have is limited by the size of
+*CMK_REFNUM_TYPE*. Specifically, no more than :math:`2^{SIZE}-1`, where :math:`SIZE`
+is the size of *CMK_REFNUM_TYPE* in bits, may be outstanding at any time.
+Waiting on more futures will cause a fatal error in non-production builds,
+and will cause the program to hang in production builds. To increase this limit,
```
Comment (rbuch) [inline doc/charm++/manual.rst:5865]: This should say something about the default type being `unsigned short` and it usually maxing out at 65,535.

**Review by rbuch — APPROVED** [review]

File: doc/charm++/manual.rst [hunk 3]
```diff
@@ -5858,6 +5858,16 @@ Other functions complete the API for futures. *CkReleaseFuture* destroys
 a future. *CkProbeFuture* tests whether the future has already finished
 computing the value of the expression.
 
+The maximum number of outstanding futures a PE may have is limited by the size of
+*CMK_REFNUM_TYPE*. Specifically, no more than :math:`2^{SIZE}-1`, where :math:`SIZE`
```
Comment (minitu) [inline doc/charm++/manual.rst:5862]:
```suggestion
*CMK_REFNUM_TYPE*. Specifically, no more than :math:`2^{SIZE}-1` futures, where :math:`SIZE`
```

File: doc/charm++/manual.rst [hunk 4]
```diff
@@ -5858,6 +5858,16 @@ Other functions complete the API for futures. *CkReleaseFuture* destroys
...
+is the size of *CMK_REFNUM_TYPE* in bits, may be outstanding at any time.
+Waiting on more futures will cause a fatal error in non-production builds,
+and will cause the program to hang in production builds. The default *CMK_REFNUM_TYPE*
+is ``unsigned short``, limiting each PE to 65,535 outstanding futures.
+To increase this limit, build Charm++ with a larger *CMK_REFNUM_TYPE*, e.g., specifying
```
Comment (minitu) [inline doc/charm++/manual.rst:5867]:
```suggestion
To increase this limit, build Charm++ with a larger *CMK_REFNUM_TYPE*, e.g. specifying
```

File: doc/charm++/manual.rst [hunk 5]
```diff
@@ -5858,6 +5858,16 @@ Other functions complete the API for futures. *CkReleaseFuture* destroys
...
+``--with-refnum-type=uint`` when building Charm++.
```
Comment (minitu) [inline doc/charm++/manual.rst:5868]:
```suggestion
``--with-refnum-type=uint`` to use ``unsigned int`` when building Charm++.
```

**Review by matthiasdiener — APPROVED** [review]

**Review by minitu — APPROVED** [review]

**Pull request merged**